src/libraries/System.Private.Runtime.InteropServices.JavaScript/tests/System/Runtime/InteropServices/JavaScript/DelegateTests.cs

```diff
@@ -0,0 +1,214 @@
+            Runtime.InvokeJS(@"
+                var del = App.call_test_method (""CreateDelegateMethod"", [  ]);
+                del(""Hic sunt dracones"");
+            ");
+
+            Assert.Equal("Hic sunt dracones", HelperMarshal._delMethodResultValue);
+        }
+
+        [Fact]
+        public static void MarshalAnonDelegateMethod()
+        {
+            HelperMarshal._delAnonMethodStringResultValue = string.Empty;
+            Runtime.InvokeJS(@"
+                var del = App.call_test_method (""CreateDelegateAnonMethodReturnString"", [  ]);
+                del(""Hic sunt dracones"");
+            ");
+
+            Assert.Equal("Notification received for: Hic sunt dracones", HelperMarshal._delAnonMethodStringResultValue);
+        }
+
+
+        [Fact]
+        public static void MarshalLambdaDelegateMethod()
+        {
+            HelperMarshal._delLambdaMethodStringResultValue = string.Empty;
+            Runtime.InvokeJS(@"
+                var del = App.call_test_method (""CreateDelegateLambdaMethodReturnString"", [  ]);
+                del(""Hic sunt dracones"");
+            ");
+
+            Assert.Equal("Notification received for: Hic sunt dracones", HelperMarshal._delLambdaMethodStringResultValue);
+        }
+
+        [Fact]
+        public static void MarshalDelegateMethodReturnString()
+        {
+            HelperMarshal._delMethodStringResultValue = string.Empty;
+            Runtime.InvokeJS(@"
+                var del = App.call_test_method (""CreateDelegateMethodReturnString"", [  ]);
+                var res = del(""Hic sunt dracones"");
+                App.call_test_method (""SetTestString1"", [ res ]);
+            ");
+
+            Assert.Equal("Received: Hic sunt dracones", HelperMarshal._delMethodStringResultValue);
+        }
+
+        [Fact]
+        public static void MarshalCustomMultiDelegateAcceptingString()
+        {
+            HelperMarshal._custMultiDelStringResultValue = string.Empty;
+            Runtime.InvokeJS(@"
+                var del = App.call_test_method (""CreateCustomMultiDelegateAcceptingString"", [  ]);
+                del(""Moin"");
+            ");
+
+            Assert.Equal("  Hello, Moin!  GoodMorning, Moin!", HelperMarshal._custMultiDelStringResultValue);
+        }
+
+        [Fact]
+        public static void MarshalCustomMultiActionAcceptingString()
+        {
+            HelperMarshal._custMultiActionStringResultValue = string.Empty;
+            Runtime.InvokeJS(@"
+                var del = App.call_test_method (""CreateCustomMultiActionAcceptingString"", [  ]);
+                del(""MoinMoin"");
+            ");
+
+            Assert.Equal("  Hello, MoinMoin!  GoodMorning, MoinMoin!", HelperMarshal._custMultiActionStringResultValue);
+        }
+
+        [Theory]
+        [MemberData(nameof(Object_Prototype))]
+        public static void MarshalFunctionAcceptingUint8Array(Function objectPrototype)
```

Comment:
Should other argument types be tested too?

Reply:
Will add more

Reply:
Added in recent commits